lambdas/gro-convert-to-json/src/test/java/uk/gov/di/data/lep/GroConvertToJsonTest.java

```diff
@@ -328,6 +359,20 @@ void convertToJsonThrowsMappingExceptionWhenExpectedRecordsButFoundNone() {
         httpClientMock.close();
     }
 
+    @Test
+    void convertToJsonThrowsMappingExceptionWhenExpectedRecordCountDoesntMatchFoundRecordCount() {
+        var httpClientMock = mockStatic(HttpClient.class);
+        httpClientMock.when(HttpClient::newHttpClient).thenReturn(httpClient);
+        when(awsService.getFromBucket(anyString(), anyString())).thenReturn(mockS3objectResponseExpectedRecordCountNotMatchingFoundRecordCount);
+
+        var exception = assertThrows(MappingException.class, () -> underTest.handleRequest(event, context));
+
+        assertTrue(exception.getMessage().contains("Expected 5 records but 2 were found"));
```

Comment:
we should be able to get a bit more out of the assertion here, e.g.
```java
assertThat(exception.getMessage()).contains("");
```

will give us a clearer error if this fails